synth.py

```diff
@@ -170,6 +170,48 @@
 if count != 2:
     raise Exception("Too many or too few replacements in pull() methods.")
 
+# Silence deprecation warnings in pull() method flattened parameter tests.
+s.replace(
+    "tests/unit/gapic/pubsub_v1/test_subscriber.py",
+    "import mock",
+    "\g<0>\nimport warnings",
+)
+count = s.replace(
+    "tests/unit/gapic/pubsub_v1/test_subscriber.py",
+    textwrap.dedent(
+        r"""
+        ([^\n\S]+# Call the method with a truthy value for each flattened field,
+        [^\n\S]+# using the keyword arguments to the method\.)
+        \s+(client\.pull\(.*?\))"""
+    ),
+    """\n\g<1>
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=DeprecationWarning)
+            \g<2>""",
+    flags = re.MULTILINE | re.DOTALL,
+)
+
+if count < 1:
+    raise Exception("Catch warnings replacement failed.")
+
+count = s.replace(
+    "tests/unit/gapic/pubsub_v1/test_subscriber.py",
+    textwrap.dedent(
+        r"""
+        ([^\n\S]+# Call the method with a truthy value for each flattened field,
+        [^\n\S]+# using the keyword arguments to the method\.)
+        \s+response = (await client\.pull\(.*?\))"""
+    ),
+    """\n\g<1>
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=DeprecationWarning)
+            \g<2>""",
+    flags = re.MULTILINE | re.DOTALL,
+)
+
```

Comment:
I don't see the deprecation warnings being checked, as requested in https://github.com/googleapis/python-pubsub/issues/367,

Reply:
We have this aspect covered in two existing tests ([one](https://github.com/googleapis/python-pubsub/blob/c707f81fdadd118bd2bc48aebe74d453f35cad0d/tests/unit/pubsub_v1/subscriber/test_subscriber_client.py#L210-L224), [two](https://github.com/googleapis/python-pubsub/blob/c707f81fdadd118bd2bc48aebe74d453f35cad0d/tests/unit/pubsub_v1/subscriber/test_subscriber_client.py#L227-L247)), which is why they don't show up here in the diff.

On the other hand, the tests patched with this replacement focus on something else and the warnings produced by them are merely a side effect, thus we just silence them.